paf-mvp-core-js/tests/test-cases/node.test.ts

```diff
@@ -0,0 +1,196 @@
+import { IJsonValidator, JsonSchemaType, JsonValidation } from '@core/validation/json-validator';
+import { IdentityConfig, Node } from '@core/express';
+import { createRequest, createResponse, MockResponse } from 'node-mocks-http';
+import { NextFunction, Response } from 'express';
+import { NodeError, NodeErrorType } from '@core/errors';
+import { getTimeStampInSec } from '@core/timestamp';
+import { encodeBase64, QSParam } from '@core/query-string';
+
+const buildStaticJsonValidator = (validationResult: boolean): IJsonValidator => {
+  const jsonValidation: JsonValidation = {
+    isValid: validationResult,
+    value: {},
+    errors: validationResult ? [] : [new Error('error message from validator')],
+  };
+  return {
+    start: jest.fn(),
+    validate: jest.fn((schema: JsonSchemaType, jsonStr: string) => jsonValidation),
+  };
+};
+const identity: IdentityConfig = {
+  type: 'operator',
+  // Name of the OneKey participant
+  name: 'Example operator',
+  // Current public key
+  publicKeys: [
+    {
+      // Timestamps are expressed in seconds
+      startTimestampInSec: getTimeStampInSec(new Date('2022-01-01T10:50:00.000Z')),
+      endTimestampInSec: getTimeStampInSec(new Date('2022-12-31T12:00:00.000Z')),
+      publicKey: '',
+    },
+  ],
+  // Email address of DPO
+  dpoEmailAddress: 'contact@example.onekey.network',
+  // URL of a privacy page
+  privacyPolicyUrl: new URL('https://example.onekey.network/privacy'),
+};
+describe('Json body validator handler', () => {
+  let response: MockResponse<Response>;
+  let nextFunction: NextFunction;
+  const payload = { some_key: 'some_text' };
+  const request = createRequest({
+    method: 'POST',
+    body: payload,
+  });
+  beforeEach(() => {
+    response = createResponse();
+    nextFunction = jest.fn();
+  });
+
+  test('should pass an InvalidJsonBody error to the nextFunction if the validation of request body fails', () => {
+    const mockJsonValidatorAlwaysKO = buildStaticJsonValidator(false);
+    const validationSpy = jest.spyOn(mockJsonValidatorAlwaysKO, 'validate');
+    const node = new Node('MyNode', identity, mockJsonValidatorAlwaysKO, () => Promise.resolve({ verify: () => true }));
```

Comment:
Can you define a mock above for the last parameter of the Node constructor ?
By reading the instance, it is not clear that we mock a `PublicKeyProvider`.

FAR NIT BTW do you know why we have the types `PublicKeys` (note the plural) and `PublicKeyProvider` ? It seems to be the same thing.

Reply:
I added the mock.

For the second point, I think it's the same thing. Something is wrong with the current implementation.
I will investigate in depth and correct it if needed in a later commit.

Reply:
After deeper looking, `PublicKeys` and `PublicKeyProvider` are not the same thing.
Public keys are the node's public keys (as all nodes must implement the identityEndpoint interface).
PublicKeyProvider is the key provider used by the node to verify request signature and content.